src/net.cpp

```diff
@@ -1114,7 +1144,11 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     // on all platforms.  Set it again here just to be sure.
     SetSocketNoDelay(hSocket);
 
-    if (IsBanned(addr) && !whitelisted)
+    int bannedlevel = IsBannedLevel(addr);
+
+    // Don't accept connections from banned peers, but if our inbound slots aren't almost full, accept
+    // if the only banning reason was an automatic misbehavior ban.
+    if (!whitelisted && bannedlevel > (nInbound + 1 < nMaxInbound))
```

Comment:
nit: `bannedlevel > (nInbound + 1 < nMaxInbound)` is not the most readable code, but I do appreciate the comment.